### ISSUE TITLE
chaining request with Promise.all returns array of Promises instead of results

### ISSUE THREAD

**Issue by JacobZ** [dmcquay/node-apac#68]
Hi,

I am currently try to write a server which can be requested with a query. The server then should make an ItemSearch on amazon and request all pages that are available (up to 10) and finally return all responses bundeled to the client. Since ```execute()``` supports Promises i tried doing it with ```Promise.all()```. Unfortunately in ```Promise.all().then((resultArray) => {})``` the resultArray contains only functions (probably promises) instead of the actual data.

here is my code:
```javascript
var Amazon = require('apac').OperationHelper;

var client = new Amazon({
		  locale: 'DE',
		  awsId: "<awsID>",
		  awsSecret: "<awsSecret>",
		  assocId: "42",
		  maxRequestsPerSecond: 1,
		  xml2jsOptions:  {
		  	explicitArray: false
		  }
		}, {locale: 'DE'});

var getStuff = function(theTitle) {
	var query = {
		Title: theTitle,
		SearchIndex: 'Books',
		BrowseNodeId: '698198',
		Power: 'binding:not (Kindle or Kalender)',
		Sort: '-publication_date',
		ResponseGroup: 'ItemAttributes,Images',
		ItemPage: 1
	};
	return makeRequests('ItemSearch', query);
};

var makeRequests = function(theMethod, theQuery) {
	return client.execute(theMethod, theQuery).then(function(theResult) {
		var pageCount = theResult.result.ItemSearchResponse.Items.TotalPages;
		var requests = [];
		if(pageCount > 10) pageCount = 10;			
		for(var i = 2; i < pageCount; i++) {
			var query = Object.assign({}, theQuery);
            query.ItemPage = i;
			requests.push(client.execute(theMethod, query));
		}
		return Promise.all(requests).then(function(theResults) {
			var data = theResults[0];
			console.log(theResults);
            for(var i = 1; i < theResults.length; i++) {            	
                var items = theResults[i].result.ItemSearchResponse.Items.Item;
                data.result.ItemSearchResponse.Items.Item = data.result.ItemSearchResponse.Items.Item.concat(items);
            }
            return data;
		});
	});

};


getStuff('Bakuman').then(function(result) {
	console.log(result);
})
.catch(function(err) {
	console.log(err);
});
```

**Comment by dmcquay** [comment]
That code looks correct. I'll have to experiment when I get a chance to see
if there's a bug (in your code or mine).

**Comment by JacobZ** [comment]
have you had the chance to look at it yet?

best 
Jacob

**Comment by dmcquay** [comment]
I found the issue and it is indeed a bug in node-apac. I'll get a fix out today.

**Comment by dmcquay** [comment]
I just released 2.0.1 which resolves this issue. Thank you for reporting it!
